src/besseli.jl

```diff
@@ -130,6 +130,209 @@ function besseli1x(x::T) where T <: Union{Float32, Float64}
     return z
 end
 
+#####
+#####  Implementation for complex arguments
+#####
+
+function besseli0(z::ComplexF64)
+    isconj = false
+    if real(z) < 0.0
+        z = -z
+    end
+    if imag(z) < 0.0
+        z = conj(z)
+        isconj = true
+    end
+    if abs(z) > 17.5
+        # use asymptotic expansion for large arguments
+        zinv = 1 / z
+        e = exp(z)
+        sinv =sqrt(zinv)/sqrt(2*pi)
+        p = evalpoly(zinv*zinv, (1.0, 0.0703125, 0.112152099609375, 0.5725014209747314, 6.074042001273483, 110.01714026924674, 3038.090510922384, 118838.42625678325, 6.252951493434797e6, 4.259392165047669e8, 3.646840080706556e10, 3.8335346613939443e12, 4.8540146868529006e14, 7.286857349377656e16))
+        p2 = evalpoly(zinv*zinv, (0.125, 0.0732421875, 0.22710800170898438, 1.7277275025844574, 24.380529699556064, 551.3358961220206, 18257.755474293175, 832859.3040162893, 5.0069589531988926e7, 3.8362551802304335e9, 3.6490108188498334e11, 4.218971570284097e13, 5.827244631566907e15, 9.47628809926011e17))
+        r = e * sinv * muladd(p2, zinv, p) + im * muladd(p2, -zinv, p) * sinv / e
+    elseif real(z) < 4.5
+        # use taylor series around the roots (slight offset) of J0(z)
+        # use relation I0(z) = J0(im*z)
+        _z = imag(z) + abs(real(z))*im
+        if real(_z) < 2.2
```

Comment:
you should check that LLVM is smart enough to nest these branches in smart ways or generating a jump table.

Reply:
It's kinda a messy implementation 🤷‍♂️ I think there's a better way to express this. I'll work a little more on this!